tests/ui/deriving/deriving-smart-pointer-neg.rs

```diff
@@ -35,6 +35,13 @@ struct NotTransparent<'a, #[pointee] T: ?Sized> {
     ptr: &'a T,
 }
 
+#[derive(SmartPointer)]
+#[repr(transparent)]
+struct NoMaybeSized<'a, #[pointee] T> {
+    //~^ ERROR: SmartPointer` is meaningless because `T` is not `?Sized
+    ptr: &'a T,
+}
```

Comment:
Should we have a test that `?std::marker::Sized` succeeds?

Reply:
Oops I somehow pushed this commit. :sweat: I intend to open another PR.

It is working. To verify it, the test has updated with sensible ways that `Sized` marker can be used in.